_episodes/32-software-design.md

```diff
@@ -1,19 +1,26 @@
 ---
-title: "Software Design"
+title: "Software Architecture and Design"
 teaching: 25
-exercises: 20
+exercises: 25
 questions:
 - "Why should we invest time in software design?"
 - "What should we consider when designing software?"
+- "What is software architecture?"
 objectives:
+- "List the common aspects of software design."
+- "Describe the term technical debt and how it impacts software."
 - "Understand the goals and principles of designing 'good' software."
-- "Understand code decoupling and code abstraction design techniques."
-- "Understand what code refactoring is."
+- "Use a diagramming technique to describe a software architecture."
+- "What are the components of Model-View-Controller (MVC) architecture?"
+- "Understand the use of common design patterns to improve the extensibility, reusability and 
+overall quality of software."
+- "List some best practices when designing software."
 keypoints:
 - "'Good' code is designed to be maintainable: readable by people who did not author the code, 
 testable through a set of automated tests, adaptable to new requirements."
-- "The sooner you adopt a practice of designing your software in the lifecycle of your project, 
-the easier the development and maintenance process will."
+- "Use abstraction and decoupling to logically separate the different aspects of your software within design as well as implementation."
+- "Use refactoring to improve existing code to improve it's consistency internally and within its overall architecture."
```

Comment:
```suggestion
- "Use refactoring to improve existing code to improve it's consistency internally and within its overall architecture."
```
```suggestion
- "Use refactoring to improve existing code to improve its consistency internally and within its overall architecture."
```

Reply:
Do we need the keypoints on abstractions, decoupling and refactoring here? They are the topics of the following two episodes and have been addressed there. Maybe we should just check if they are alined with my branch https://github.com/carpentries-incubator/python-intermediate-development/pull/327? No harm in repeating either - just a thought.